File: android/src/main/java/com/lightningkite/khrysaliswebsite/layouts/FaqXml.kt
//
// FaqXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.khrysaliswebsite.layouts

import android.widget.*
import android.view.*
import com.lightningkite.khrysalis.views.*
import com.lightningkite.khrysaliswebsite.R

class FaqXml {

    lateinit var filter: EditText
    lateinit var questions: com.lightningkite.khrysalis.views.android.VerticalRecyclerView
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.faq, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        filter = view.findViewById<EditText>(R.id.filter)
        questions = view.findViewById<com.lightningkite.khrysalis.views.android.VerticalRecyclerView>(R.id.questions)
        
        
        
        return view
    }
}

File: web/src/layout/FaqXml.ts
//
// FaqXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId } from 'khrysalis/dist/views/html'
import { customViewSetDelegate } from 'khrysalis/dist/views/CustomView.actual'
import htmlForDefault from './faq.html'
//! Declares com.lightningkite.khrysaliswebsite.layouts.FaqXml
export class FaqXml {
    xmlRoot!: HTMLElement;
    filter!: HTMLInputElement;
    questions!: HTMLDivElement;
    loadHtmlString(): string {
        return htmlForDefault;
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString());
        this.xmlRoot = view
        this.filter = getViewById<HTMLInputElement>(view, "filter");
        this.questions = getViewById<HTMLDivElement>(view, "questions");
        return view
    }
}
